src/utilsplitstring.h

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2018 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <iterator>
+#include <set>
+#include <string>
+#include <vector>
+#include <type_traits>
+
+#ifndef BITCOIN_UTILSPLITSTRING_H
+#define BITCOIN_UTILSPLITSTRING_H
+
+/**
+ * Tokenizes a string by any of the given separators.
+ * @param[in] tokens            The container (either an instance of std::vector<std::string>
+ *                              or std::set<std::string>) to add tokenized string parts to.
+ * @param[in] str               The string to tokenize.
+ * @param[in] anyOfSeparator    A string with valid separators.
+ * @param[in] mergeEmpty        Set to true to merge adjacent separators (empty tokens); otherwise false (default).
+ */
+template<typename ContainerT>
+void Split(ContainerT& tokens, const std::string& str, const std::string& anyOfSeparator, bool mergeEmpty = false)
+{
+    static_assert(
+        std::is_same<std::vector<std::string>, ContainerT>::value ||
+        std::is_same<std::set<std::string>, ContainerT>::value,
+        "`Split` has only been tested with template arguments of type `std::vector<std::string>` and `std::set<std::string>`");
+
+    auto insertIt = std::inserter(tokens, tokens.end());
+    if (str.empty()) {
+        *insertIt = "";
+        return;
+    }
+
+    const auto begin = str.cbegin();
+    const auto end = str.cend();
+
+    for (auto it = begin; it < end; ++it) {
+        bool foundSeparator = false;
+        auto start = it;
+        while (it < end && !(foundSeparator = anyOfSeparator.find(*it) != std::string::npos)) {
```

Comment:
Try to reformulate without mixing assignment and comparison. That would make it easier to read and reason about :-)

Reply:
Hey, thanks @practicalswift! Yes, absolutely. Please see 3704125.